### PR TITLE
[REVIEW] Updating Dask-XGBoost to accept Dask-cuDF Objects

### PR DESCRIPTION
**Pull request by mtjrider**

### CODE REVIEW

File: dask_xgboost/core.py [hunk 1]
```diff
@@ -68,7 +67,7 @@ def concat(L):
     elif isinstance(L[0], (gd.DataFrame, gd.Series)):
         return gd.concat(L)
     elif isinstance(L[0], xgb.DMatrix):
-        return L
+        return
```
Comment (jakirkham) [inline dask_xgboost/core.py:70]: Was the `return L` meant to be dropped here?

Comment (mtjrider) [reply dask_xgboost/core.py:70]: Yes. Technically, this logical block doesn't do anything. The `data` is tested under `isinstance(data, xgb.DMatrix)` in the `train_part` function, and when it passes, no `concat` is done. This makes it explicit that `concat(xgb.DMatrix)` should never be called.

File: dask_xgboost/core.py [hunk 2]
```diff
@@ -95,20 +94,24 @@ def train_part(env, param, list_of_parts, dmatrix_kwargs=None, **kwargs):
...
             dmatrix_kwargs = {}
         dtrain = xgb.DMatrix(data, labels, **dmatrix_kwargs)
 
     elif labels[0] is None and isinstance(data[0], xgb.DMatrix):
         dtrain = data[0]
-        #dmatrix_kwargs["feature_names"] = getattr(data, 'columns', None)
         if dmatrix_kwargs is None:
             dmatrix_kwargs = {}
 
+    elif labels[0] is None and isinstance(data[0], gd):
```
Comment (randerzander) [inline dask_xgboost/core.py]: Should it be:
elif labels[0] is None and isinstance(data[0], gd.DataFrame):

?

Comment (mtjrider) [reply dask_xgboost/core.py]: Corrected.

**Review by randerzander — APPROVED** [review]
This worked for me :1st_place_medal:

**Pull request merged**